### PR TITLE
Refactor `Reauthenticate` components to handle generic MFA challenges.

### PR DESCRIPTION
**Pull request by Joerger**
The overarching change in this PR is the ability of all `Reauthenticate` flows to handle generic MFA challenges (sso, webauthn, otp) rather than handling each type differently or not at all. 

Note: `useReauthenticate` should now be the goto state handler method for reauthentication flows. 

Changes:
- Refactor `useReauthenticate` (used in the `Reauthenticate` components).
  - Return a `submitWithMFA` method instead of `submitWithWebAuthn` and `submitWithOTP`. SSO MFA will use `submitWithMFA('sso')`.
  - Deprecate `OnAuthenticated` prop
  - Hold MFA challenge in state - prevents the current behavior of retrieving a new MFA challenge every time you open the component or switch between available options. We only need a fresh MFA challenge after the current challenge has been used up.
  - Use update MFA Options from https://github.com/gravitational/teleport/pull/49678 to derive reauth and register options from MFA challenge ^
- Update change/add device wizards according to `useReauthenticate` refactors.
- Use `useReauthenticate` for change password wizard to get the same changes described above.
- Update `createPrivilegeToken` endpoint to accept generic MFA response instead of TOTP or Webauthn, so it will now support SSO MFA.

Note on `/e`: This PR is dependent on the [e PR](https://github.com/gravitational/teleport.e/pull/5656) and vice versa. Therefore I'll need to merge this PR with e PR as the head of `/e`. Then I'll make a follow up to update the e ref to master once the e PR is merged, shortly after this PR is merged.

TODO (follow ups):
  - (Optional) Instead of getting a privilege token before adding/deleting a device, just use the mfa response directly. 
    - This is a piece of work which has been forgotten after being made possible with https://github.com/gravitational/teleport/pull/32271
    - Note to self: we will still need to get a privilege token if the user clicks the otp register option since it would cash in the mfa response for the qr code, and as a result the user would need to reauthenticate to switch to webauthn. Also failure/cancel on the last step may need to go back to initial reauth step.

Prerequisite for supporting SSO MFA in device management (add/delete), connection tester (discover), change password, and account recovery flows.

Depends on https://github.com/gravitational/teleport/pull/49678,  https://github.com/gravitational/teleport/pull/49679, https://github.com/gravitational/teleport.e/pull/5656

### CODE REVIEW

File: web/packages/teleport/src/Account/Account.test.tsx [hunk 1]
```diff
@@ -189,6 +192,14 @@ test('password change', async () => {
   const ctx = createTeleportContext();
   ctx.storeUser.setState({ passwordState: PasswordState.PASSWORD_STATE_UNSET });
   jest.spyOn(ctx.mfaService, 'fetchDevices').mockResolvedValue([]);
+  jest.spyOn(auth, 'getMfaChallenge').mockResolvedValue({
+    webauthnPublicKey: {} as PublicKeyCredentialRequestOptions,
```
Comment (avatus) [inline web/packages/teleport/src/Account/Account.test.tsx:196]: Shouldn't this be null? I think if its not returned during the challenge its null

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/Account/Account.test.tsx (hunk 1 above)
Comment (Joerger) [reply web/packages/teleport/src/Account/Account.test.tsx:196]: We want it to be returned during the challenge here, so it needs to be non-null

Comment (avatus) [reply web/packages/teleport/src/Account/Account.test.tsx:196]: Oh duh, that makes sense

**Review by avatus — APPROVED** [review]

File: web/packages/teleport/src/Account/Account.test.tsx (hunk 1 above)
Comment (Joerger) [reply web/packages/teleport/src/Account/Account.test.tsx:196]: Eh, it's not necessary for this test which just cares about option selection, some other tests mock it. Maybe we'll need to add in the future though.

**Comment by Joerger** [comment]
@ryanclark @avatus I added one more commit if you don't mind taking a look e60ec3d494a114ad48ff865ead3eb3526ff80199

**Review by avatus — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Pull request merged**